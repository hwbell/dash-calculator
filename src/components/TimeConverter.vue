<template>
  <div class="time-converter-component top-left-all-col">
    <div id="current-unix-time" class="top-left-all-row row">
      <div class="col-sm-4" style="padding: 0">
        <strong>Unix Timestamp:</strong>
      </div>

      <input class="unix-time-input col-sm-8" type="text" :value="unixTime" />
    </div>
    <div id="current-verbal-time" class="top-left-all-row row">
      <p class="col-sm-4" style="font-size: 16pt;">
        <strong>Regular Time & Date:</strong>
      </p>
      <div class="col-sm-8">
        <div class="row" style="margin-right: 20px">
          <div
            v-for="(prop, ind) in this.regularTime"
            v-bind:key="ind"
            class="time-input-holder col-xs-6 col-sm-4 col-md-2 col-lg-2"
          >
            <p>{{ prop.text }}</p>
            <input
              class="regular-time-input"
              type="text"
              :name="prop.text"
              v-model="prop.val"
              @input="handleRegularTimeChange(prop)"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
// import ExpressionWindow from './ExpressionWindow';

export default {
  name: "TimeConverter",
  // props: {
  // msg: String,
  // },
  // components: {ExpressionWindow},
  created: function() {
    const moment = require("moment");

    this.unixTime = moment().unix();
    this.regularTime = [
      {
        text: "Month",
        val: moment().format("M"),
      },
      {
        text: "Day",
        val: moment().format("D"),
      },
      {
        text: "Year",
        val: moment().format("YYYY"),
      },
      {
        text: "Hour",
        val: moment().format("hh"),
      },
      {
        text: "Minute",
        val: moment().format("mm"),
      },
      {
        text: "Second",
        val: moment().format("ss"),
      },
    ];
    setInterval(() => {
      // console.log("interval");
      this.timeDate = moment().format("MMMM Do YYYY");
      this.timeHours = moment().format("h:mm a");
    }, 1000);
  },
  data: function() {
    return {
      message: "This is a TimeConverter component for us to start with!",
      regularTime: [],
      unixTime: "",
    };
  },
  computed: {
    // a computed getter
    reversedMessage: function() {
      // `this` points to the vm instance
      return this.message
        .split("")
        .reverse()
        .join("");
    },
    regularTimeKeys: function() {
      return Object.keys(this.regularTime);
    },
  },
  methods: {
    greet: function() {
      // `this` inside methods point to the Vue instance
      this.message = "Nice to meet you!";
    },
    handleRegularTimeChange: function(prop) {
      // console.log(prop);
      this.regularTime.forEach((obj, i) => {
        if (this.regularTime[i].text === prop.text) {
          this.regularTime[i].val = prop.val;
          return console.log(this.regularTime);
        }
      })
    },
    resetTime: function() {
      const moment = require("moment");
      this.unixTime = moment().unix();
      this.regularTime = [
        {
          text: "Month",
          val: moment().format("M"),
        },
        {
          text: "Day",
          val: moment().format("D"),
        },
        {
          text: "Year",
          val: moment().format("YYYY"),
        },
        {
          text: "Hour",
          val: moment().format("hh"),
        },
        {
          text: "Minute",
          val: moment().format("mm"),
        },
        {
          text: "Second",
          val: moment().format("ss"),
        },
      ];
    },
  },
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h3 {
  margin: 40px;
}
.time-converter-component {
  min-width: 400px;
  color: white;
}
#current-unix-time {
  width: 100%;
  margin: 20px 0;
  font-size: 16pt;
  color: white;
}
#current-verbal-time {
  width: 100%;
  /* margin: 20px; */
}
.time-input-holder {
  font-size: 14pt;
}
.regular-time-input {
  border-radius: 4px;
  width: 100px;
}
.unix-time-input {
  border-radius: 12px;
  max-width: 200px;
}
</style>
