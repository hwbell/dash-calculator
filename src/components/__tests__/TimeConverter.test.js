import { mount } from '@vue/test-utils'
import TimeConverter from '../TimeConverter.vue'
import Vue from 'vue';
const message = 'This is a TimeConverter component for us to start with!';

describe('TimeConverter', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = mount(TimeConverter)
  })

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  })

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  })
  
  test('renders a button for each object in the array map', () => {
    expect(wrapper.findAll('#time-converter-component').length).toBe(1);
    expect(wrapper.findAll('#current-unix-time').length).toBe(1);
    expect(wrapper.findAll('#current-verbal-time').length).toBe(1);
  })

  test('assigns the time to the input values on created', async () => {
    await Vue.nextTick();

    expect(wrapper.vm.regularTime).toBeTruthy();
    expect(wrapper.vm.regularTime.length).toBe(6);
    wrapper.vm.regularTime.forEach((input) => {
      expect(typeof(input.val)).toBe('string');
      expect(typeof(input.text)).toBe('string');
    })

  })

  // test(`assigns the computed value on the component properly`, () => {
  //   let revMessage = message.split('').reverse().join('');
  //   expect(wrapper.vm.reversedMessage).toBe(revMessage);
  // })  

  // test('should change the message when the greet method is called', () => {
  //   wrapper.vm.greet();
  //   expect(wrapper.vm.message).toBe('Nice to meet you!');
  // })


})