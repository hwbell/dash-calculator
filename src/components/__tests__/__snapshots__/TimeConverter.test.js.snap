// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeConverter renders correctly 1`] = `
<div
  class="time-converter-component top-left-all-col"
>
  <h3>
    Time Converter
  </h3>
   
  <div
    class="space-all-row row"
    id="current-unix-time"
  >
    <div
      class="left-all-col"
    >
      <p
        style="margin: 0px;"
      >
        <strong>
          Unix Timestamp:
        </strong>
      </p>
       
      <input
        class="unix-time-input"
        type="text"
      />
       
      <button
        class="btn btn-sm btn-primary"
      >
        
        sync time & date
      
      </button>
    </div>
  </div>
   
  <hr />
   
  <div
    class="space-all-row row"
    id="current-verbal-time"
  >
    <div
      class="left-all-col"
    >
      <p
        style="margin: 0px;"
      >
        <strong>
          Time & Date:
        </strong>
      </p>
       
      <div
        class="row"
        style="margin-right: 20px;"
      >
        <div
          class="time-input-holder col-xs-12 col-sm-12 col-md-6 col-lg-4"
        >
          <p
            style="margin: 0px;"
          >
            <strong>
              Month
            </strong>
          </p>
           
          <input
            class="regular-time-input"
            name="Month"
            type="text"
          />
        </div>
        <div
          class="time-input-holder col-xs-12 col-sm-12 col-md-6 col-lg-4"
        >
          <p
            style="margin: 0px;"
          >
            <strong>
              Day
            </strong>
          </p>
           
          <input
            class="regular-time-input"
            name="Day"
            type="text"
          />
        </div>
        <div
          class="time-input-holder col-xs-12 col-sm-12 col-md-6 col-lg-4"
        >
          <p
            style="margin: 0px;"
          >
            <strong>
              Year
            </strong>
          </p>
           
          <input
            class="regular-time-input"
            name="Year"
            type="text"
          />
        </div>
        <div
          class="time-input-holder col-xs-12 col-sm-12 col-md-6 col-lg-4"
        >
          <p
            style="margin: 0px;"
          >
            <strong>
              Hour
            </strong>
          </p>
           
          <input
            class="regular-time-input"
            name="Hour"
            type="text"
          />
        </div>
        <div
          class="time-input-holder col-xs-12 col-sm-12 col-md-6 col-lg-4"
        >
          <p
            style="margin: 0px;"
          >
            <strong>
              Minute
            </strong>
          </p>
           
          <input
            class="regular-time-input"
            name="Minute"
            type="text"
          />
        </div>
        <div
          class="time-input-holder col-xs-12 col-sm-12 col-md-6 col-lg-4"
        >
          <p
            style="margin: 0px;"
          >
            <strong>
              Second
            </strong>
          </p>
           
          <input
            class="regular-time-input"
            name="Second"
            type="text"
          />
        </div>
      </div>
       
      <button
        class="btn btn-sm btn-primary"
      >
        
        sync unix timestamp
      
      </button>
    </div>
  </div>
</div>
`;
